x/axelarnet/types/genesis.go

```diff
@@ -57,6 +60,34 @@ func (m GenesisState) Validate() error {
 		return getValidateError(sdkerrors.Wrapf(err, "invalid transfer queue state"))
 	}
 
+	// ibc transfer ID should be unique
+	transferSeen := make(map[nexus.TransferID]bool)
+	for _, t := range m.IBCTransfers {
+		if transferSeen[t.ID] {
+			return getValidateError(fmt.Errorf("duplicate transfer ID %d", t.ID))
+		}
+
+		if t.Status == TransferNonExistent {
+			return getValidateError(fmt.Errorf("invalid status of transfer %s", t.ID))
+		}
+
+		if err := t.ValidateBasic(); err != nil {
+			return getValidateError(sdkerrors.Wrapf(err, "invalid transfer %s", t.ID))
+		}
+
+		transferSeen[t.ID] = true
+	}
+
+	// IBCTransfer ID should be uniquely mapped
+	transferIDSeen := make(map[uint64]bool)
+	for seqKey, id := range m.SeqIDMapping {
```

Comment:
Do we need to also make sure that all `.IBCTransfers` have a sequence here set in the map? Also, I think this needs to be deterministic, otherwise different node may return a different error at different iteration.

Reply:
good point, updated. We only have the seq -> id mapping for pending transfer, we removed when transfers completed/ failed.

Reply:
Still non-deterministic, right?

Reply:
forgot to delete the old